src/new/wallet.js

```diff
@@ -1279,25 +1305,19 @@ class HathorWallet extends EventEmitter {
    * @memberof HathorWallet
    * @inner
    **/
-  async getPreProcessedData(key) {
+  getPreProcessedData(key) {
     if (Object.keys(this.preProcessedData).length === 0) {
-      await this.preProcessWalletData();
+      throw new Error('Wallet data has not been processed yet');
     }
     return this.preProcessedData[key];
   }
 
   setState(state) {
-    if (state === HathorWallet.READY && state !== this.state) {
-      // Became ready now, so we prepare new localStorage data
-      // to support using this facade interchangable with wallet service
-      // facade in both wallets
-
-      // Notice: preProcessWalletData is currently async because `getTxBalance`
-      // is async but it resolves instantly -- we only changed `getTxBalance` to async
-      // to match signatures with the new facade. If we ever need to do something really
-      // async on preProcessWalletData and wait for it before setting state to READY, we
-      // probably should refactor setState to be async
-      this.preProcessWalletData();
+    if (state === HathorWallet.PROCESSING && state !== this.state) {
```

Comment:
I understand the previous code had state transition logic here. I'm not sure this is the best place to add have it, though. For now, maybe we should move this code to a "onEnterStateProcessing()" method?

Reply:
Done

Reply:
Great! Thanks!